### PR TITLE
chore(deps): bump openresty from 1.21.4.1 to 1.21.4.2

### PR DESCRIPTION
**Pull request by bungle**
### Summary

See: https://openresty.org/en/ann-1021004002.html

KAG-2140

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -163,6 +163,8 @@
 
 ### Dependencies
 
+- Bumped OpenResty from 1.21.4.1 to 1.21.4.2
```
Comment (chronolaw) [inline CHANGELOG.md:166]: The change log entry should be added in a new  `unreleased` section after 3.4 release.

File: build/openresty/patches/lua-cjson-2.1.0.10_01-empty_array.patch [hunk 2]
```diff
@@ -1,12 +0,0 @@
-diff -ruN a/bundle/lua-cjson-2.1.0.8/lua_cjson.c b/bundle/lua-cjson-2.1.0.8/lua_cjson.c
---- a/bundle/lua-cjson-2.1.0.10/lua_cjson.c	2022-01-11 15:11:17.495464192 +0800
```
Comment (chronolaw) [inline build/openresty/patches/lua-cjson-2.1.0.10_01-empty_array.patch:2]: This bug has been fixed in https://github.com/openresty/lua-cjson/pull/82, I think that we should remove this patch.

Comment (bungle) [reply build/openresty/patches/lua-cjson-2.1.0.10_01-empty_array.patch:2]: @chronolaw that is what the PR does. It removes that patch.

File: CHANGELOG.md (hunk 1 above)
Comment (bungle) [reply CHANGELOG.md:166]: @chronolaw, I agree on that, thus I guess we need to wait for release, a merge to changelog from release, and then add a new section. Or we just continue normally like here, and resolve this issue when we merge changelog from 3.4 to master.

**Comment by chronolaw** [comment]
Need to fix `dyn_upstream_keepalive.patch`.

**Comment by bungle** [comment]
> Need to fix `dyn_upstream_keepalive.patch`.

Yes, if it is because of it. Currently I cannot reproduce with my setup, and patch applies cleanly. But I can take another look at it. Perhaps things works differently on Linuxes.

**Comment by fffonion** [comment]
I rebased the patch by using `patch` tool and generate the patch again. The patch implementation for Bazel is `patch` and somtimes it fails to apply for patches that are applied with fuzz
```
patching file ngx_lua-0.10.25/src/ngx_http_lua_balancer.c
Hunk #3 succeeded at 238 with fuzz 2.
Hunk #4 succeeded at 303 (offset 10 lines).
Hunk #5 succeeded at 354 (offset 10 lines).
Hunk #6 succeeded at 376 (offset 10 lines).
Hunk #7 succeeded at 415 (offset 10 lines).
Hunk #8 succeeded at 452 (offset 10 lines).
Hunk #9 succeeded at 490 (offset 10 lines).
Hunk #10 succeeded at 938 (offset 10 lines).
Hunk #11 succeeded at 952 (offset 10 lines).
Hunk #12 succeeded at 965 (offset 10 lines).
Hunk #13 succeeded at 997 (offset 10 lines).
Hunk #14 succeeded at 1020 (offset 10 lines).
Hunk #15 succeeded at 1032 (offset 10 lines).
Hunk #16 succeeded at 1107 (offset 10 lines).
Hunk #17 succeeded at 1138 (offset 10 lines).
Hunk #18 succeeded at 1195 (offset 10 lines).
Hunk #19 succeeded at 1224 (offset 10 lines).
Hunk #20 succeeded at 1250 (offset 10 lines).
Hunk #21 succeeded at 1278 (offset 10 lines).
patching file ngx_lua-0.10.25/src/ngx_http_lua_common.h
Hunk #1 succeeded at 247 with fuzz 2 (offset 7 lines).
Hunk #2 succeeded at 341 with fuzz 1 (offset 20 lines).
patching file ngx_lua-0.10.25/src/ngx_http_lua_module.c
Hunk #1 succeeded at 1158 with fuzz 1 (offset 41 lines).
```

**Review by chronolaw — APPROVED** [review]

**Comment by bungle** [comment]
> I rebased the patch by using `patch` tool and generate the patch again. The patch implementation for Bazel is not `patch` and somtimes it fails to apply for patches that are applied with fuzz

Yes, it turned out that Mac patch tool was not complaining, so I installed `brew install gpatch` and could reproduce. Thanks for figuring it out, I fixed rest of the fuzz. Should be no ok, but lets see.

**Comment by fffonion** [comment]
@bungle The above error is because lua-resty-healthcheck uses `sock:tlshandshake` that is from our patch but is removed in this PR. Is this intentional and should we update lua-resty-healthcheck to use the new API or add the cosocket-mtls patch back?

**Comment by bungle** [comment]
> @bungle The above error is because lua-resty-healthcheck uses `sock:tlshandshake` that is from our patch but is removed in this PR. Is this intentional and should we update lua-resty-healthcheck to use the new API or add the cosocket-mtls patch back?

@fffonion good catch! I tried to search for `tlshandshake` but didn't find anything, so I think it would be best to update `lua-resty-healthcheck` to use `sslhandshake`.

**Pull request merged**